docs/src/man/split_apply_combine.md

```diff
@@ -2,7 +2,20 @@
 
 Many data analysis tasks involve splitting a data set into groups, applying some functions to each of the groups and then combining the results. A standardized framework for handling this sort of computation is described in the paper "[The Split-Apply-Combine Strategy for Data Analysis](http://www.jstatsoft.org/v40/i01)", written by Hadley Wickham.
 
-The DataFrames package supports the Split-Apply-Combine strategy through the `by` function, which takes in three arguments: (1) a DataFrame, (2) one or more columns to split the DataFrame on, and (3) a function or expression to apply to each subset of the DataFrame.
+The DataFrames package supports the Split-Apply-Combine strategy through the `by` function, which takes in three arguments: (1) a `DataFrame`, (2) one or more columns to split the `DataFrame` on, and (3) a specification of one or more functions to apply to each subset of the `DataFrame`. This specification can be of the following forms:
+- a `col => function` pair indicating that `function` should be called with the vector of values for column `col`, which can be a column name or index
+- a `cols => function` pair indicating that `function` should be called with a named tuple holding columns `cols`, which can be a tuple or vector of names or indices
+- a (named) tuple or vector of such pairs, producing each a single separate column
+- a function which will be called with a `SubDataFrame` corresponding to each group; this form should be avoided due to its poor performance
+
+In all of these cases, the function can return either a single row or multiple rows, with a single or multiple columns:
```

Comment:
It would be good to be precise in which cases only a single column is allowed.

If a single column is allowed then I understand that any of the return-value options is valid as long as it produces one column. Right?

Reply:
In this case (single column) - what do we do if there is a name clash, i.e. the returned value returns a single column e.g. as a `DataFrame` with one column having a different name than indicated in the call.

Reply:
Right. Currently the only way to indicate a name in the call is to pass a named tuple of pairs, and in that case we only allow returning a single value or vector, so that there are no names which could clash. I've added a mention of this.

Reply:
OK